packages/react/src/migrations/update-15-6-3/webpack-config-setup.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  formatFiles,
+  logger,
+  readProjectConfiguration,
+  Tree,
+  updateProjectConfiguration,
+} from '@nrwl/devkit';
+import { forEachExecutorOptions } from '@nrwl/workspace/src/utilities/executor-options-utils';
+import { basename } from 'path';
+
+export default async function (tree: Tree) {
+  forEachExecutorOptions(
+    tree,
+    '@nrwl/webpack:webpack',
+    (options: {}, projectName, targetName, _configurationName) => {
+      // If isolatedConfig is set, we don't need to do anything
+      // If it is NOT React, we don't need to do anything
+      if (
+        options?.['isolatedConfig'] ||
+        !(
+          options?.['main']?.match(/main\.(t|j)sx$/) ||
+          options?.['webpackConfig'] === '@nrwl/react/plugins/webpack'
+        )
+      ) {
+        return;
+      }
+
+      // If webpackConfig is set, update it with the new options
+      // If webpackConfig is not set, we need to create a new
+      // webpack.config.js file and set the path to it in the
+      // executor options
+
+      if (
+        options?.['webpackConfig'] &&
+        options['webpackConfig'] !== '@nrwl/react/plugins/webpack'
+      ) {
+        let oldName = options['webpackConfig'];
+        if (options['webpackConfig'].endsWith('.js')) {
+          oldName = options['webpackConfig'].replace('.js', '.old.js');
+        }
+        if (options['webpackConfig'].endsWith('.ts')) {
+          oldName = options['webpackConfig'].replace('.ts', '.old.ts');
+        }
+
+        const justTheFileName = basename(options['webpackConfig']);
+        renameFile(tree, options['webpackConfig'], oldName);
+
+        tree.write(
+          options['webpackConfig'],
+          `
+            const { composePlugins, withNx } = require('@nrwl/webpack');
+            const { withReact } = require('@nrwl/react');
+
+            // Nx plugins for webpack.
+            module.exports = composePlugins(withNx(), withReact(), (config, { options, context }) => {
+              // Note: This was added by an Nx migration.
+              // You should consider inlining the logic into this file.
+              return require('./${justTheFileName}')(config, context);
```

Comment:
should this be `oldName`? The original file is overwritten.

Reply:
Yes!!!! I saw that too, oops!!! 😅

Reply:
The issue is actually above, on the line where I'm setting the `justTheFileName`. It's using the `webpackConfig` path, not the `oldName` to get the file name. I fixed that line,

Reply:
oh i get it, sorry i totally misread